pmda/test/test_util.py

```diff
@@ -130,3 +131,55 @@ def test_make_balanced_slices_ValueError(n_frames, n_blocks,
     with pytest.raises(ValueError):
         make_balanced_slices(n_frames, n_blocks,
                              start=start, stop=stop, step=step)
+
+
+@pytest.mark.parametrize('n_frames', [3, 4, 10, 19, 101, 331, 1000])
+def test_second_order_moments(n_frames):
+    # generate array of random positions in range [-100, 100] for 100 time steps
+    pos = 200*(np.random.random(size=(n_frames, 1000, 3)) - 0.5)
+    # random splitting point
+    isplit = (np.random.randint(1, n_frames-1))
+    # split into two partitions; should be fixed to split by n_blocks partitions
+    p1, p2 = pos[:isplit], pos[isplit:]
+    # create [t, mu, M] lists
+    S1 = [len(p1), p1.mean(axis=0), sumofsquares(p1)]
+    S2 = [len(p2), p2.mean(axis=0), sumofsquares(p2)]
+    # run lists through second_order_moments
+    result = second_order_moments(S1, S2)
+    # compare result to calculations over entire pos array
+    assert result[0] == len(pos)
+    assert_almost_equal(result[1], pos.mean(axis=0))
+    assert_almost_equal(result[2], sumofsquares(pos))
+
+
+@pytest.mark.parametrize('n_blocks', [2, 3, 4, 5, 10, 100, 500])
+@pytest.mark.parametrize('n_frames', [1000, 10000, 50000, 100000])
+def test_reduce(n_frames, n_blocks):
+    # generate array of random positions in range [-100, 100] for 100 time steps
+    pos = 200*(np.random.random(size=(n_frames, 1000, 3)) - 0.5).astype(np.float64)
```

Comment:
You can make this a [fixture](https://docs.pytest.org/en/latest/fixture.html)
```python
@pytest.fixture(scope="module")
def pos():
   return 200*(np.random.random(size=(n_frames, 1000, 3)) - 0.5).astype(np.float64)
```

and then 
```
@pytest.mark.parametrize('n_blocks', [2, 3, 4, 5, 10, 100, 500])
@pytest.mark.parametrize('n_frames', [1000, 10000, 50000, 100000])
def test_reduce(pos, n_frames, n_blocks):
   ...
```

With `scope="module"` the fixture only runs once and is reduced everywhere where you use `pos` as an argument for a test. Without the `scope` keyword or `scope="function"` (I think, check the docs), each test will get a new version of `pos`.